include/nuttx/arch.h

```diff
@@ -930,6 +930,43 @@ int up_copy_section(FAR void *dest, FAR const void *src, size_t n);
 #  define up_getpicbase(ppicbase)
 #endif
 
+/****************************************************************************
+ * Percpu support
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: up_update_task
+ *
+ * Description:
+ *   We can utilize percpu storage to hold information about the
+ *   current running task. If we intend to implement this feature, we would
+ *   need to define two macros that help us manage this percpu information
+ *   effectively.
+ *
+ *   up_this_task: This macro is designed to read the contents of the percpu
+ *                 register to retrieve information about the current
+ *                 running task.This allows us to quickly access
+ *                 task-specific data without having to disable interrupts,
+ *                 access global variables and obtain the current cpu index.
+ *
+ *   up_update_task: This macro is responsible for updating the contents of
+ *                   the percpu register.It is typically called during
+ *                   initialization or when a context switch occurs to ensure
+ *                   that the percpu register reflects the information of the
+ *                   newly running task.
+ *
+ * Input Parameters:
+ *   current tcb
+ *
+ * Returned Value:
+ *   current tcb
+ *
+ ****************************************************************************/
+
+#ifndef up_update_task
```

Comment:
It seems this macro is implemented in the next patch, I suggest squashing them as they are the same feature.

Reply:
each arch need provides the different implementation and mayn't provide an implementation at all, so it's better that the first patch only introduces the hook point.

Reply:
Ok, makes sense